Index: ghc-llvm-next/ghc-src/configure.ac
===================================================================
--- ghc-llvm-next.orig/ghc-src/configure.ac
+++ ghc-llvm-next/ghc-src/configure.ac
@@ -551,7 +551,7 @@ esac
 # tools we are looking for. In the past, GHC supported a number of
 # versions of LLVM simultaneously, but that stopped working around
 # 3.5/3.6 release of LLVM.
-LlvmVersion=3.9
+LlvmVersion=4.0
 AC_SUBST([LlvmVersion])
 sUPPORTED_LLVM_VERSION=$(echo \($LlvmVersion\) | sed 's/\./,/')
 AC_DEFINE_UNQUOTED([sUPPORTED_LLVM_VERSION], ${sUPPORTED_LLVM_VERSION}, [The supported LLVM version number])
