Index: ghc-llvm-head/ghc-src/compiler/llvmGen/Llvm/PpLlvm.hs
===================================================================
--- ghc-llvm-head.orig/ghc-src/compiler/llvmGen/Llvm/PpLlvm.hs
+++ ghc-llvm-head/ghc-src/compiler/llvmGen/Llvm/PpLlvm.hs
@@ -73,7 +73,7 @@ ppLlvmGlobal (LMGlobal var@(LMGlobalVar
             Nothing -> empty
 
         rhs = case dat of
-            Just stat -> ppr stat
+            Just stat -> pprSpecialStatic stat
             Nothing   -> ppr (pLower $ getVarType var)
 
         -- Position of linkage is different for aliases.
Index: ghc-llvm-head/ghc-src/compiler/llvmGen/Llvm/Types.hs
===================================================================
--- ghc-llvm-head.orig/ghc-src/compiler/llvmGen/Llvm/Types.hs
+++ ghc-llvm-head/ghc-src/compiler/llvmGen/Llvm/Types.hs
@@ -175,6 +175,13 @@ instance Outputable LlvmStatic where
   ppr (LMSub s1 s2)
       = pprStaticArith s1 s2 (sLit "sub") (sLit "fsub") "LMSub"
 
+
+pprSpecialStatic :: LlvmStatic -> SDoc
+pprSpecialStatic (LMBitc v t) =
+    ppr (pLower t) <> text ", bitcast (" <> ppr v <> text " to " <> ppr t <> char ')'
+pprSpecialStatic stat = ppr stat
+
+
 pprStaticArith :: LlvmStatic -> LlvmStatic -> LitString -> LitString -> String -> SDoc
 pprStaticArith s1 s2 int_op float_op op_name =
   let ty1 = getStatType s1
